### PR TITLE
Baseline/entropy

### PR DESCRIPTION
**Pull request by paulnovello**
Entropy method. Just a score based on $-\sum p_i log(p_i)$ seen in [Likelihood Ratios for Out-of-Distribution Detection](https://proceedings.neurips.cc/paper/2019/hash/1e79596878b2320cac26dd792a6c51c9-Abstract.html

### CODE REVIEW

**Review by y-prudent — COMMENTED** [review]
Great (and efficient) work :) LGTM

File: oodeel/methods/entropy.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    As always, training data is expected to have lower score than OOD data.
+    """
+
+    def __init__(self):
+        super().__init__(output_layers_id=[-1])
+
+    def _score_tensor(self, inputs: TensorType) -> np.ndarray:
+        """
+        Computes an OOD score for input samples "inputs" based on
+        energy, namey $-logsumexp(logits(inputs))$.
```
Comment (cofri) [inline oodeel/methods/entropy.py:55]: Docstring is the one of Energy method

File: docs/notebooks/tensorflow/demo_entropy_tf.ipynb [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+                "\n",
+                "The method consists in using the Entropy of the input data computed using the Entropy\n",
+                "$\\sum_{c=0}^C p(y=c| x) \\times log(p(y=c | x))$ where\n",
+                "$p(y=c| x) = \\text{model}(x)$.\n",
+                "\n",
+                "Here, we focus on a toy convolutional network trained on MNIST[0-4] and a ResNet model\n",
+                "trained on CIFAR-10, respectively challenged on MNIST[5-9] and SVHN OOD datasets.\n",
+                "\n",
+                "**Reference**\n",
+                "[Entropy-based Out-of-distribution Detection](http://arxiv.org/abs/2010.03759),\n",
```
Comment (cofri) [inline docs/notebooks/tensorflow/demo_entropy_tf.ipynb:20]: Is it the right reference?

File: docs/notebooks/tensorflow/demo_entropy_tf.ipynb [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "# Entropy method\n",
+                "\n",
+                "This notebook aims at evaluating the **Entropy method**.\n",
+                "\n",
+                "The method consists in using the Entropy of the input data computed using the Entropy\n",
+                "$\\sum_{c=0}^C p(y=c| x) \\times log(p(y=c | x))$ where\n",
+                "$p(y=c| x) = \\text{model}(x)$.\n",
```
Comment (cofri) [inline docs/notebooks/tensorflow/demo_entropy_tf.ipynb:14]: Maybe give more precision on p(y|x)? For example, softmax(model(x)) ?

**Review by cofri — APPROVED** [review]
One more method 💪
